tailwind.config.js

```diff
@@ -43,5 +61,50 @@ module.exports = {
 
       addUtilities(utilities, variants("borderColor"));
     },
+    plugin(function ({ addVariant, e }) {
+      addVariant("aria-selected", ({ modifySelectors, separator }) => {
+        modifySelectors(({ className }) => {
+          return `.${e(
+            `aria-selected${separator}${className}`,
+          )}[aria-selected="true"]`;
+        });
+      });
+    }),
+    plugin(function ({ addVariant, e }) {
+      addVariant("aria-disabled", ({ modifySelectors, separator }) => {
+        modifySelectors(({ className }) => {
+          return `.${e(
+            `aria-disabled${separator}${className}`,
+          )}[aria-disabled="true"]`;
+        });
+      });
+    }),
+    plugin(function ({ addVariant, e }) {
+      addVariant("is-range-selection", ({ modifySelectors, separator }) => {
+        modifySelectors(({ className }) => {
+          return `.${e(
+            `is-range-selection${separator}${className}`,
+          )}[data-is-range-selection]`;
+        });
+      });
+    }),
+    plugin(function ({ addVariant, e }) {
+      addVariant("is-range-start", ({ modifySelectors, separator }) => {
+        modifySelectors(({ className }) => {
+          return `.${e(
+            `is-range-start${separator}${className}`,
+          )}[data-is-selection-start]`;
+        });
+      });
+    }),
+    plugin(function ({ addVariant, e }) {
+      addVariant("is-range-end", ({ modifySelectors, separator }) => {
+        modifySelectors(({ className }) => {
+          return `.${e(
+            `is-range-end${separator}${className}`,
+          )}[data-is-selection-end]`;
+        });
+      });
+    }),
```

Comment:
Of course, we can simplify this into a single plugin !
Left it separately for experimenting, we can simplify them at last IMO